public/app/features/alerting/unified/components/receivers/useAlertmanagerConfigHealth.ts

```diff
@@ -0,0 +1,38 @@
+import { AlertmanagerConfig, Route } from '../../../../../plugins/datasource/alertmanager/types';
+
+export type ContactPointConfigHealth = 'OK' | 'Unused';
+
+export interface AlertmanagerConfigHealth {
+  contactPoints: Record<string, ContactPointConfigHealth>;
+}
+
+export function useAlertmanagerConfigHealth(config: AlertmanagerConfig): AlertmanagerConfigHealth {
+  if (!config.receivers) {
+    return { contactPoints: {} };
+  }
+
+  if (!config.route) {
+    return { contactPoints: Object.fromEntries(config.receivers.map((r) => [r.name, 'Unused'])) };
```

Comment:
A bit confused about this part, when is there no route defined? Is a default policy not a mandatory route?

Reply:
According to our TS types definition, it's not required, hence the code. According to the Alertmanager docs, the `route` parameter is optional so it seems to be correct

Reply:
Interesting! I do wonder what happens if you boot Alertmanager with no default policy – but that's irrelevant here.